python/cudf/cudf/dataframe/column.py

```diff
@@ -48,14 +48,14 @@ def _concat(cls, objs, dtype=None):
         from cudf.dataframe.categorical import CategoricalColumn
 
         if len(objs) == 0:
-            if pd.api.types.is_categorical_dtype(dtype):
+            if dtype == np.dtype("object"):
+                return StringColumn(data=nvstrings.to_device([]), null_count=0)
+            elif pd.api.types.is_categorical_dtype(dtype):
```

Comment:
Use reference equality comparison here instead against the CategoricalDtypeType.

Reply:
@kkraus14 Made the changes, Ready for review.